### PR TITLE
fix(form-field): autofill background overrides error, disabled and overlay states (#DS-4096)

### PR DESCRIPTION
**Pull request by lskramarov**

### CODE REVIEW

File: packages/components/form-field/form-field-control.ts [hunk 1]
```diff
@@ -36,6 +37,15 @@ export abstract class KbqFormFieldControl<T> {
     /** Whether the control is in an error state. */
     readonly errorState: boolean;
 
+    /**
+     * Whether the control's value was filled in by the browser.
+     *
+     * Implement it only on controls that can actually be autofilled — text inputs and textareas. It is a
+     * signal rather than a plain property because the form field reads it from a host binding and runs
+     * `OnPush`: a signal read there marks the form field dirty on its own.
```
[the remark this answers is not in the document]
Comment (lskramarov) [reply packages/components/form-field/form-field-control.ts]: Reworded: it now says to implement it on a control that is itself a text input or textarea, **or** on a wrapper that forwards the state of the input it hosts, naming `KbqTagList` as that case.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: packages/components/form-field/form-field.en.md [hunk 2]
```diff
@@ -102,6 +102,37 @@ import { kbqFormFieldDefaultOptionsProvider } from '@koobiq/components/form-fiel
 })
 ```
 
+### Autofill
+
+When the browser fills a field in, `<kbq-form-field>` gets the `kbq-form-field_autofilled` class and the field is tinted with `--kbq-form-field-states-autofill-background`. The browser's own background and text color are painted over, so an autofilled field looks the same in every browser.
+
+The tint is the weakest state: `focused`, an error, `disabled` and `inOverlay` all win over it. `noBorders` is not a state and does not — it only makes the border transparent, so an autofilled `noBorders` field keeps the tint. The state is detected through the CDK's `AutofillMonitor`, so it is also readable from code — `KbqInput`, `KbqInputPassword`, `KbqTextarea` and `KbqTagInput` (forwarded by `KbqTagList`) expose an `autofilled` signal:
```
[the remark this answers is not in the document]
Comment (lskramarov) [reply packages/components/form-field/form-field.en.md]: Rewritten as "`noBorders` does not win over it, because it is not a state — it only makes the border transparent"; the RU mirror got the same fix.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: packages/components/textarea/e2e.ts [hunk 3]
```diff
@@ -67,6 +67,37 @@ import { KbqTextareaModule } from '@koobiq/components/textarea';
             <textarea kbqTextarea placeholder="Placeholder" [canGrow]="true" [(ngModel)]="longValue"></textarea>
             <kbq-hint>Hint</kbq-hint>
         </kbq-form-field>
+
+        <!--
+            Autofill is faked through the class AutofillMonitor drives at runtime — ':autofill' cannot be
```
[the remark this answers is not in the document]
Comment (lskramarov) [reply packages/components/textarea/e2e.ts]: Reworded to "the class that `AutofillMonitor` adds at runtime".

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: packages/components/tags/e2e.ts [hunk 4]
```diff
@@ -259,6 +259,25 @@ export class E2eTagListStates {}
                 <kbq-cleaner />
             </kbq-tag-list>
         </kbq-form-field>
+
+        <!--
+            Autofill is faked through the class AutofillMonitor drives at runtime — ':autofill' cannot be
```
[the remark this answers is not in the document]
Comment (lskramarov) [reply packages/components/tags/e2e.ts]: Reworded to "the class that `AutofillMonitor` adds at runtime".

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: packages/components/input/e2e.ts [hunk 5]
```diff
@@ -184,7 +188,15 @@ export class E2eInputStateAndStyle {
         [{ state: ['focused'] }],
         [{ state: ['error'] }],
         [{ state: ['error', 'focused'] }],
-        [{ state: ['disabled'] }]
+        [{ state: ['disabled'] }],
+        // Autofill is faked through `kbq-form-field_autofilled`, the class `AutofillMonitor` drives at
```
[the remark this answers is not in the document]
Comment (lskramarov) [reply packages/components/input/e2e.ts]: Reworded to "the class that `AutofillMonitor` adds at runtime".

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: packages/components/form-field/_form-field-theme.scss [hunk 6]
```diff
@@ -15,14 +15,26 @@
...
+        // background on the control with a UA `!important` rule, and the mask has to repaint whatever
+        // background *this* state resolved to. Reading the winner back through a variable is what keeps
+        // the mask from ever drifting away from the container — including for `_in-overlay`, `_no-borders`
+        // and consumer token overrides. Declared in the same rule as the `background-color` that reads
+        // it, so one cascade decides both and they cannot drift apart.
+        --kbq-form-field-current-background: var(--kbq-form-field-#{$state-name}-background);
+
         border-color: var(--kbq-form-field-#{$state-name}-border-color);
-        background-color: var(--kbq-form-field-#{$state-name}-background);
+        background-color: var(--kbq-form-field-current-background);
```
Comment (NikGurev) [inline packages/components/form-field/_form-field-theme.scss]: Фон должен меняться сразу при вставке значения и фон инпута не должен меняться , чтобы не было наложения цвета заднего фона с форм филдом

https://github.com/user-attachments/assets/42269327-3eef-4eca-b29c-c7ec90f2fb5e

Comment (NikGurev) [reply packages/components/form-field/_form-field-theme.scss]: Скриншот тест не добавляли, потому что была проблема с autofill в playwright, надо проверить

Comment (lskramarov) [reply packages/components/form-field/_form-field-theme.scss]: Оба замечания приняты, поправил в `c5d0ebb3`.

**Наложение.** Токен подсветки полупрозрачный (`--kbq-background-theme-less` — 10% в светлой теме, 21% в тёмной), а я красил им и контейнер, и сам инпут (`box-shadow: inset 0 0 0 40rem`). Цвет накладывался дважды — прямоугольник инпута темнее, чем отступы формфилда вокруг него. Хуже того, полупрозрачная заливка всё равно не скрывала фон браузера, он просвечивал. Убрал закраску инпута совсем: контрол остаётся прозрачным, фон браузера подавляется бесконечным `transition-duration` на `background-color` (переходы в каскаде стоят выше important-UA), подсветку рисует только контейнер. Перекрашивается лишь текст — через `-webkit-text-fill-color`, иначе браузер навязывает свой цвет; он теперь берётся из текущего состояния, поэтому у disabled/невалидного поля цвет текста правильный.

**Задержка.** Подсветка была завязана на класс `kbq-form-field_autofilled`, который `AutofillMonitor` вешает только после `animationstart` и прохода change detection. Добавил вторую ветку селектора — `:has(:is(.kbq-input, .kbq-tag-input, .kbq-textarea):is(:autofill, :-webkit-autofill))`: она срабатывает в том же проходе стилей, в котором браузер подставляет значение, так что фон меняется вместе с текстом. Класс оставил — он даёт сигнал `autofilled` в TS и единственный способ проверить состояние в e2e, так как `:autofill` синтетически не воспроизводится. Обе ветки задают одно и то же, поэтому поздняя ничего не перерисовывает.

Заодно убрал компенсацию геометрии (прозрачный бордер + padding): она была нужна только чтобы непрозрачная заливка не перекрывала рамку фокуса, а теперь перекрывать нечем.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

Comment (lskramarov) [reply packages/components/form-field/_form-field-theme.scss]: Проверил — оказалось, что тест сделать можно. Добавил в `d79cc5a4`.

**Почему раньше не получалось.** CDP-домена `Autofill` в хромиуме, который поставляется с Playwright, просто нет: `Schema.getDomains` возвращает 35 доменов, `Autofill` среди них отсутствует, поэтому `Autofill.enable` / `setAddresses` / `trigger` падают с "wasn't found" — домен собирается только в Chrome-branded сборках, а в нашем docker-образе брендированного Chrome нет. То есть настоящее автозаполнение действительно не воспроизвести.

**Что работает.** `CSS.forcePseudoState` с `forcedPseudoClasses: ['autofill']` — псевдокласс форсится на этапе разрешения стилей, `:has()` на этом инвалидируется, `Element.matches(':autofill')` его видит. Обёртка — `e2eForceAutofill()` в `packages/e2e/utils/autofill.ts`.

Причём этого хватает и на подавление фона: при форсированном псевдоклассе браузер применяет и свой фон тоже, так что `background-color` контрола читается как `rgb(232, 240, 254)` с alpha 0 — то есть ровно то, что делает наш бесконечный `transition-duration`. Убери его — alpha станет 1 и тест упадёт.

Добавил 4 теста в `input/e2e.playwright-spec.ts`: ветка `:has()` красит поле так же, как класс; фон браузера держится на нулевой alpha и на контроле ничего не рисуется (это прямая защита от наложения из прошлого замечания); autofill проигрывает error и disabled. Цвета сравниваются с тем же полем до форсирования, а не захардкожены, так что тесты переживут смену токенов. Мутационно проверил: убрал ветку `:has()` — падает первый тест, убрал `transition-duration` — падает второй. Скриншотные бейзлайны не изменились.

Две ловушки записал в комментарии к хелперу: `session.detach()` молча снимает форсирование (первый заход у меня из-за этого дал три зелёных теста, которые ничего не проверяли), и `-webkit-autofill` как имя псевдокласса не принимается — только стандартное `autofill`.

Чего этот подход не даёт: значение не подставляется и `animationstart` не стреляет, поэтому `AutofillMonitor` молчит и класс `kbq-form-field_autofilled` не появляется — эта половина по-прежнему проверяется подделкой класса в сетке.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

**Comment by lskramarov** [comment]
Закрыто в пользу https://github.com/koobiq/angular-components/pull/1908
